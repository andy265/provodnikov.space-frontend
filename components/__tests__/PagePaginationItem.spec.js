const { createRenderer } = require('vue-server-renderer')
const { mount, shallow } = require('vue-test-utils')
const PagePaginationItem = require('~/components/PagePaginationItem')

describe('PagePaginationItem.vue', () => {
  it('has same HTML structure', () => {
    let wrapper = shallow(PagePaginationItem, {
      propsData: {
        isLink: false,
        to: '/page'
      },
      slots: {
        default: '<span>next page</span>'
      }
    })
    let renderer = createRenderer()
    // eslint-disable-next-line
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('can have link', () => {
    let wrapper = mount(PagePaginationItem, {
      propsData: {
        isLink: true,
        to: '/page'
      }
    })
    expect(wrapper.find('a').hasAttribute('href', '/page')).toBeTruthy()
  })
})
