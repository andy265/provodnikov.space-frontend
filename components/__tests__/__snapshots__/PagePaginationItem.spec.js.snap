// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagePaginationItem.vue has same HTML structure 1`] = `<div class="page-pagination-item"><span>next page</span></div>`;
